### PR TITLE
[WIP] Journals feature development

### PR DESCRIPTION
**Pull request by 1aurend**
@dananjohnson to make it easier to review, I linked the breadcrumbs such that, if you start from `http://manifold.lvh/journals/journal-0/volumes/slug-volume-0` and click back, you'll see all the routes I've setup.

### CODE REVIEW

File: client/src/frontend/containers/JournalDetail/index.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  const settings = useSelectSettings();
+  const { paginationState, handlePageChange } = usePaginationState(
+    location,
+    20
+  );
+  const { filterState, updateFilterState } = useFilterState(location, {
+    collectionOrder: match.params.id
+  });
+  // TODO: add useDispatchIssues hook
+  // TODO: add useSelectIssues hook
```
Comment (1aurend) [inline client/src/frontend/containers/JournalDetail/index.js:30]: It felt like diminishing returns to keep mocking hooks when the api is a couple days away. The re-fetch on filter state change logic also needs to be added here.

File: client/src/hooks/index.js [hunk 2]
```diff
@@ -32,3 +32,17 @@ export { default as useGrabCurrentTextSection } from "./texts/use-grab-current-t
...
+export { default as useFilterState } from "./filters/use-filter-state";
+export { default as useUrlFromState } from "./url/use-url-from-state";
+
+export { default as useSelectAllIssues } from "./journals/use-select-all-issues";
+export { default as useDispatchAllIssues } from "./journals/use-dispatch-all-issues";
+export { default as useSelectAllJournals } from "./journals/use-select-all-journals";
+export { default as useDispatchAllJournals } from "./journals/use-dispatch-all-journals";
+export { default as useSelectJournal } from "./journals/use-select-journal";
+export { default as useSelectVolume } from "./journals/use-select-volume";
+export { default as useDispatchJournal } from "./journals/use-dispatch-journal";
```
Comment (1aurend) [inline client/src/hooks/index.js:48]: I think it's a good idea to add index files to each folder, but I'll do that as a separate commit at some point.

File: client/src/frontend/containers/JournalDetail/index.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    if (heroStyles?.mediumLandscape) return heroStyles.mediumLandscape;
+    return null;
+  };
+
+  return (
+    <div>
+      <CheckFrontendMode
+        debugLabel="ProjectCollectionDetail"
+        isProjectSubpage
+      />
```
Comment (dananjohnson) [inline client/src/frontend/containers/JournalDetail/index.js:61]: Check the `debugLabel` prop in the `CheckFrontendMode` component for these new routes and make sure they're accurate.

**Review by dananjohnson — CHANGES_REQUESTED** [review]
Looks great, @1aurend! 👏🏻👏🏻👏🏻  One little change, then make sure to clean up the WIP and fixup commits.

**Pull request merged**